src/js/formapp.js

```diff
@@ -145,7 +146,19 @@ async function renderForm({ definition, isReadOnly, storedSubmission, formData,
       return;
     }
 
-    router.request('submit:form', { response: extend({}, response, { data }) });
+    try {
+      const submitResponse = extend(getResponse(form, submitReducers, data), response, { data });
+
+      // Remove empty data to prevent { __empty__: true }
+      each(['fields', 'flow', 'action'], key => {
+        if (isEmpty(submitResponse[key])) delete submitResponse[key];
```

Comment:
@zfixler re https://github.com/RoundingWell/care-ops-frontend/pull/1187/files#r1394594132, it looks like the empties are removed here.